Blueprint: add section Trust Base, Proofs, Gaps and Limitations (#36)
runtimeverification/evm-equivalence: blueprint/src/chapter/axioms.tex

All such subgoals should be provable requiring at most some small tweaking of
the surrounding functions.

\section{Gas subgoal sorries}

> One special (almost systematic) kind of subgoal sorry is that of gas-related subogals.

Reviewer: ```suggestion
One special (almost systematic) kind of subgoal sorry is that of gas-related subgoals.
```

> For the current opcodes targeted for equivalence we have not prioritized (given the time constraints) proving the equiavelce of gas consuption or other gas-related obligations.

Reviewer: ```suggestion
For the current opcodes targeted for equivalence we have not prioritized (given the time constraints) proving the equivalence of gas consumption or other gas-related obligations.
```

The reason for this is that we consider gas agreeance as less critical than the other proof obligations that are covered.
In fact, if the EvmYul model allowed for a gasless mode of execution, as does KEVM, it is very probable that the first version of the equivalence proofs would have made use of it.

> Another reason for not priorizing gas-related goals is that they are often quite time consuming. A good direction of research is to apply more automatic reasoning methods such as smt-based tactics or LLMs.

Reviewer: ```suggestion
Another reason for not prioritizing gas-related goals is that they are often quite time consuming. A good direction of research is to apply more automatic reasoning methods such as SMT-based tactics or LLMs.
```
